src/backend/drm/compositor/mod.rs

```diff
@@ -2003,15 +2014,25 @@ where
         Ok(frame_reference)
     }
 
-    /// Queues the current frame for rendering.
+    /// Queues the current frame for scan-out.
+    ///
+    /// If `render_frame` has not been called prior to this function or returned no damage
+    /// this function will do nothing. In this case no vblank event will be generated and it
+    /// is the callers responsibility to re-schedule rendering. A simple strategy for frame
+    /// re-scheduling is to queue a one-shot timer that will trigger after approximately one
+    /// retrace duration.
     ///
     /// *Note*: This function needs to be followed up with [`DrmCompositor::frame_submitted`]
     /// when a vblank event is received, that denotes successful scan-out of the frame.
     /// Otherwise the underlying swapchain will eventually run out of buffers.
     ///
     /// `user_data` can be used to attach some data to a specific buffer and later retrieved with [`DrmCompositor::frame_submitted`]
     pub fn queue_frame(&mut self, user_data: U) -> FrameResult<(), A, F> {
-        self.queued_frame = self.next_frame.take().map(|state| {
+        self.queued_frame = self.next_frame.take().and_then(|state| {
+            if state.is_empty() {
+                return None;
+            }
```

Comment:
Didn't we want to make this return an error, so downstream can explicitly handle, that it needs to re-schedule?

Reply:
Initially yes, but we already ignore the frame if we have no next buffer (render_frame has not been called) which was also not documented. So I thought that might be viable alternative. But I am not biased in either direction, I am also fine with making it an error. But in that case we should also discuss making no next frame an error.

Reply:
I am for consistency and explicit-ness(?). So feel free to add both error variants. :)

Reply:
Will do! But I just realized that this could lead to an unexpected situation for downstream when switching from GbmBufferedSurface to DrmCompositor. The next frame logic was taken over from next buffer, so iirc queue_buffer also just ignores the call if there is no next buffer. Do we want to change that too?

Reply:
I think we should for consistency.